### ISSUE TITLE
LR scheduler throws warning when using scaler.step instead of optimizer.step

### ISSUE THREAD

**Issue by BramVanroy** [pytorch/pytorch#44511]
## 🐛 Bug

The documentation provides [a code snippet](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.unscale_) to use the automatic scaling and AMP along side gradient clipping. If we also add a scheduler step into the mix, after the optimizer step, PyTorch throws a UserWarning which I do not think is correct:

UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "

## To Reproduce

```python
# for ANP see https://pytorch.org/docs/stable/notes/amp_examples.html
scaler.scale(loss).backward()
# Unscales the gradients of optimizer's assigned params in-place
scaler.unscale_(optimizer)
# clip in-place
clip_grad_norm_(self.model.parameters(), 1.0)
scaler.step(optimizer)
lr_scheduler.step()
scaler.update()
```

## Expected behavior

I would expect no warning here because I would expect that scaler.step calls the optimizer.step.

## Environment

 - PyTorch Version (e.g., 1.0): 1.6
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Python version: 3.8
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: RTX 2080TI

**Comment by colesbury** [comment]
@mcarilli can you look at this?

**Comment by WZMIAOMIAO** [comment]
i got same problem.

**Comment by GeoffreyChen777** [comment]
same problem.

**Comment by codertimo** [comment]
same problem

**Comment by Kevinkevin189** [comment]
same Problem.

**Comment by exiarepairii** [comment]
same Problem.

**Comment by thebuax** [comment]
Same problem.

**Comment by akihironitta** [comment]
Any updates?
